### DISCUSSION TITLE
How To Enable SSL On Client Side In Javascript

### DISCUSSION THREAD

**Discussion by brennanmanion** [rsocket/rsocket-js#260] [General]
I have a project using a spring boot backend and a react javascript front end. I have SSL enabled in my spring boot project in the properties file. However, just changing the URL to wss in my react client side is not enough. How can I include the context and correctly tell RSocket I want a secure websocket connection?

Below is an example of my javascript client connection

```js       
         const client = new RSocketClient({
            serializers: {
                data: IdentitySerializer,
                metadata: IdentitySerializer
            },
            setup: {
                dataMimeType: APPLICATION_OCTET_STREAM.string,
                keepAlive: 10000, //ms
                lifetime: 100000, //ms
                metadataMimeType: MESSAGE_RSOCKET_ROUTING.string
            },
            transport: new RSocketWebSocketClient({
                url: wsUrl
            }, BufferEncoders),
        })
```

And here is how my spring controller looks

```java
@Controller
public class ChatController {

    @ConnectMapping
    public void connect(@Headers Map<String, Object> headers)
    {
    	System.out.println("connected!");
    }
}
```

I made a PEM file using these commands

```
keytool -genkeypair -alias rsocket -keyalg RSA -keysize 2048 -storetype PKCS12 -validity 3650 -keystore rsocket-server.p12 -storepass password
 
keytool -exportcert -alias rsocket -keystore rsocket-server.p12 -storepass password -file cert.pem
 
keytool -importcert -alias rsocket -keystore client.truststore -storepass password -file cert.pem
```

And I have them configured in my RSocket server like so

```
spring.rsocket.server.port=6565
spring.rsocket.server.mapping-path=/rsocket
spring.rsocket.server.transport=websocket
spring.rsocket.server.ssl.enabled=true
spring.rsocket.server.ssl.key-store-type=PKCS12
spring.rsocket.server.ssl.key-store=/Users/brennan/Downloads/rsocket-server.p12
spring.rsocket.server.ssl.key-store-password=password
spring.rsocket.server.ssl.key-alias=rsocket
```

Is there anyone who knows how to solve this or a possible workaround?

**Comment by viglucci** [comment]
Hey @brennanmanion, can you confirm which version of rsocket-js you are using? Looks like `0.27.0`, but just wanted to be sure. Thanks.

**Comment by brennanmanion** [comment]
Yes, it is version 0.27.0

Here is my package.json

Thanks for looking into it!

{
"name": "browser-device-orientation-event",
"version": "0.1.0",
"private": true,
"dependencies": {
***@***.***/jest-dom": "^5.16.5",
***@***.***/react": "^13.4.0",
***@***.***/user-event": "^13.5.0",
"react": "^18.2.0",
"buffer": "^6.0.3",
"react-dom": "^18.2.0",
"bootstrap": "^5.2.3",
"react-bootstrap": "^2.7.0",
"react-scripts": "5.0.1",
"rsocket-core": "^0.0.27",
"rsocket-websocket-client": "^0.0.27",
"web-vitals": "^2.1.4"
},
"scripts": {
"start": "react-scripts start",
"build": "react-scripts build",
"test": "react-scripts test",
"eject": "react-scripts eject"
},
"eslintConfig": {
"extends": [
"react-app",
"react-app/jest"
]
},
"browserslist": {
"production": [
">0.2%",
"not dead",
"not op_mini all"
],
"development": [
"last 1 chrome version",
"last 1 firefox version",
"last 1 safari version"
]
}
}

**Comment by brennanmanion** [comment]
If it is of any use at all, here is my test in java to confirm my server's SSL is working correctly 

`public class RSocketWebsocketClientSslTest {

	static 
	{
		System.setProperty("javax.net.ssl.trustStore", "/Users/brennan/Downloads/client.truststore");
		System.setProperty("javax.net.ssl.trustPassword", "password");
	}
	
	@Autowired
	private RSocket rsocket;
	
	@BeforeAll
	public void setup()
	{
		this.rsocket = RSocketConnector.create()
				.connect(WebsocketClientTransport.create(URI.create("wss://localhost:6565")))
				.block();
	}
	
	
	@Test
	public void connectionTest()
	{
		Payload payload = DefaultPayload.create("java client");
		Mono<Void> mono = this.rsocket.fireAndForget(payload);
		
		StepVerifier.create(mono).verifyComplete();
	}
}`

Without setting those properties, I get an SSL connection exception.
